torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -68,6 +69,36 @@ c10::optional<at::ScalarType> PromoteIntegralType(
                                                                    : opt_dtype;
 }
 
+// Return the upper limit for a given type. For floating point types, reutrn
+// 2^mantissa.
+int64_t GetUpperLimitForType(torch::ScalarType dtype) {
```

Comment:
I'd make this use a combination of this:

https://github.com/pytorch/xla/blob/02e270008cfabb6417e07d28d469764e3c795c8c/torch_xla/csrc/tensor_util.h#L65

And this:

https://github.com/pytorch/xla/blob/02e270008cfabb6417e07d28d469764e3c795c8c/torch_xla/csrc/helpers.h#L205

I'd also make the above like:
```c++
void CheckRangeValues(torch::ScalarType dtype, at::Scalar from, at::Scalar to);
```